packages/components/src/Simple.tsx

```diff
@@ -3,92 +3,124 @@ import {
   PenroseState,
   prepareState,
   RenderInteractive,
+  RenderStatic,
   resample,
   showError,
+  stateConverged,
+  stepState,
   stepUntilConvergence,
+  variationSeeds,
 } from "@penrose/core";
 import React from "react";
 import fetchResolver from "./fetchPathResolver";
 
 export interface ISimpleProps {
-  domainString: string;
-  substanceString: string;
-  styleString: string;
+  domain: string;
+  substance: string;
+  style: string;
   variation: string;
-  initState?: PenroseState;
+  interactive?: boolean; // considered true by default
+  animate?: boolean; // considered false by default
 }
 
-interface ISimpleState {
-  state?: PenroseState;
-}
+class Simple extends React.Component<ISimpleProps> {
+  readonly canvasRef = React.createRef<HTMLDivElement>();
+  penroseState: PenroseState | undefined = undefined;
+  timerID: number | undefined = undefined; // for animation
 
-class Simple extends React.Component<ISimpleProps, ISimpleState> {
-  public readonly canvasRef = React.createRef<HTMLDivElement>();
-  constructor(props: ISimpleProps) {
-    super(props);
-    this.state = { state: undefined };
-  }
-  getInitState = async (
-    dsl: string,
-    sub: string,
-    sty: string,
-    variation: string
-  ): Promise<PenroseState | undefined> => {
-    const compilerResult = compileTrio({
-      substance: sub,
-      style: sty,
-      domain: dsl,
-      variation,
-    });
+  compile = async (): Promise<void> => {
+    this.penroseState = undefined;
+    const compilerResult = compileTrio(this.props);
     if (compilerResult.isOk()) {
       // resample because initial sampling did not use the special sampling seed
-      const initState: PenroseState = resample(
-        await prepareState(compilerResult.value)
-      );
-      const stepped = stepUntilConvergence(initState);
+      this.penroseState = resample(await prepareState(compilerResult.value));
+    } else {
+      console.log(showError(compilerResult.error));
```

Comment:
Not in scope: should probably do better error handling like rendering errors on the canvas directly.